controllers/pattern_controller.go

```diff
@@ -505,3 +535,73 @@ func (r *PatternReconciler) actionPerformed(p *api.Pattern, reason string, err e
 	}
 	return r.onReconcileErrorWithRequeue(p, reason, err, nil)
 }
+
+// MBP-184 Implementation
+// applyPatternAppDetails retrieves the status for the Pattern applications created in ArgoCD.
+// Once the Application Status is retrieve it appends the information
+// to the array and the CR is updated.
+func (r *PatternReconciler) applyPatternAppDetails(client argoclient.Interface, input *api.Pattern) (*api.Pattern, error) {
+	output := input.DeepCopy()
+	var applicationInfo api.PatternApplicationInfo
+
+	// Buid the namespace from the Pattern ObjectMeta.Name and Spec.ClusterGroupName
+	ns := output.ObjectMeta.Name + "-" + output.Spec.ClusterGroupName
+	// oc get Applications -n [pattern-name]-hub
+	// E.G. oc get Applications -n multicloud-gitops-hub
+	applications, err := client.ArgoprojV1alpha1().Applications(ns).List(context.Background(), metav1.ListOptions{})
+
+	// Return error
+	if err != nil {
+		return output, err
+	}
+
+	// Reset the array since we are going to replace it with the new application status
+	if len(output.Status.Applications) > 0 {
+		output.Status.Applications = nil
+	}
+
+	// Loop through the Pattern Applications and append the details to the Applications array
+	for _, app := range applications.Items {
+		//fmt.Println(app.Name, " ==> [", app.Status.Health.Status, "] == [", app.Status.Sync.Status, "]")
+		// Add Application information to ApplicationInfo struct
+		applicationInfo.Name = app.Name
+		applicationInfo.AppHealthStatus = string(app.Status.Health.Status)
+		applicationInfo.AppHealthMessage = app.Status.Health.Message
+		applicationInfo.AppSyncStatus = string(app.Status.Sync.Status)
+		applicationInfo.AppReconcileTime = app.Status.ReconciledAt.String()
+		if len(app.Status.Conditions) <= 0 {
+			applicationInfo.Conditions = nil // Reset Conditions array
+		} else {
+			applicationInfo.Conditions = append(applicationInfo.Conditions, app.Status.Conditions...)
+		}
+
+		// Now let's append the Application Information
+		output.Status.Applications = append(output.Status.Applications, *applicationInfo.DeepCopy())
+	}
+
+	// Update CR for the Pattern Instance
+	if output.Status.Applications != nil {
+		// Create an empty Pattern CR
+		patternCR := &api.Pattern{}
+
+		// Get the current Pattern CR
+		_ = r.Client.Get(context.TODO(), Client.ObjectKey{
+			Namespace: "openshift-operators",
+			Name:      output.ObjectMeta.Name}, patternCR)
+
+		// Replace the array in the CR for the applications
+		// with the newly generated array
+		patternCR.Status.Applications = output.Status.Applications
+
+		// Update last step in CR
+		patternCR.Status.LastStep = "update pattern application status"
+
+		// Now let's update the CR with the application status data.
+		err = r.Client.Status().Update(context.Background(), patternCR)
+		if err != nil {
+			return output, err
+		}
```

Comment:
This seems like a sane thing to do, but is in fact a non-obvious trap for new players.
The CR should only be updated once, at the very end of the reconcile loop, to avoid problems around caching and subsequent updates being perceived as "old"